### PR TITLE
Add Ingress to trino chart

### PR DESCRIPTION
**Pull request by yanyixing**

### CODE REVIEW

File: charts/trino/values.yaml
Comment (yanyixing) [reply charts/trino/values.yaml:104]: the separate values files has beed removed

**Comment by hashhar** [comment]
That values merging has been merged in https://github.com/trinodb/charts/pull/41 since it's already ready.

Can you just keep the ingress changes in this PR and I'll merge then. Please also don't bump the chart version yet, we'll do that when planning a release.

**Comment by yanyixing** [comment]
@hashhar updated

**Review by hashhar — APPROVED** [review]
Thank you.

I'll merge once CLA is processed. It's a partially manual process and is processed in batches weekly.

**Comment by yanyixing** [comment]
@hashhar The CLA is done, maybe you can merge this pr

**Comment by hashhar** [comment]
@cla-bot check

**Comment by hashhar** [comment]
@yanyixing Can you rebase this PR with something like `git rebase --reset-author main`? One of the commits has a different author so that CLA check is not passing yet.

**Comment by yanyixing** [comment]
@hashhar  Thanks for you reminder, the commit is rebased

File: charts/trino/templates/ingress.yaml
Comment (hashhar) [inline charts/trino/templates/ingress.yaml:38]: indentation issue.

File: charts/trino/templates/ingress.yaml
Comment (hashhar) [inline charts/trino/templates/ingress.yaml:26]: did you mean values.ingress.tls.secretName?

File: charts/trino/templates/ingress.yaml
Comment (hashhar) [inline charts/trino/templates/ingress.yaml:5]: why not use `networking.k8s.io/v1` available from 1.19-0? It's been available for quite a while.

File: charts/trino/templates/ingress.yaml
Comment (hashhar) [inline charts/trino/templates/ingress.yaml:38]: Also is this correct? I thought it's `service.name` as in:

```yaml
service:
  name: ...
```

Comment (hashhar) [reply charts/trino/templates/ingress.yaml:38]: Also, pathType is required AFAIK.

**Comment by hashhar** [comment]
sorry I missed couple things during review.

Another question, does the ingress terminate TLS? If so then we might want to set `http-server.process-forwarded` config so that Trino actually accepts connections that were initiated over TLS.

File: charts/trino/templates/ingress.yaml
Comment (yanyixing) [reply charts/trino/templates/ingress.yaml:26]: yes

File: charts/trino/templates/ingress.yaml
Comment (yanyixing) [reply charts/trino/templates/ingress.yaml:5]: I will update this

File: charts/trino/templates/ingress.yaml
Comment (yanyixing) [reply charts/trino/templates/ingress.yaml:38]: fullName is a variable which is refer to the `trino.fullname`, and the service name is also `trino.fullname`, so this will be ok

**Comment by yanyixing** [comment]
Ingress can passthrough ssl
If the ingress type is Nginx ingress, add the annotation `nginx.ingress.kubernetes.io/ssl-passthrough: "true"` to the ingress

File: charts/trino/templates/ingress.yaml
Comment (yanyixing) [reply charts/trino/templates/ingress.yaml]: fixed

**Comment by yanyixing** [comment]
@hashhar plz review again

**Comment by saikiran2603** [comment]
@hashhar  Any chance of this PR being merged soon ? We are waiting for this change and another one regarding authentication, in the next helm chart release ..

**Comment by lichkain** [comment]
@hashhar Hope that this PR can be merged soon! I was about to use this helm chart until I noticed it was missing configuration options for ingress.

**Comment by lichkain** [comment]
> This helm chart looks like it gets updated and has an ingress in case anyone is interested https://artifacthub.io/packages/helm/valeriano-manassero/trino

I saw that chart too, but since I started using this chart I was hoping that Ingress support could be quickly added. Especially since this PR has been pretty much ready to merge for the last few months...

**Comment by cccs-tom** [comment]
I want to chime in my support for merging this @hashhar. Our team is in the process of migrating to this chart because we were under the impression that it is Trino's "official" chart. Is that assumption incorrect?

**Review by cccs-tom — APPROVED** [review]

**Comment by cccs-tom** [comment]
Adding my review here - I agree that using `http-server.process-forwarded` or SSL pass-through is going to be deployment-specific and the chart should be flexible enough to allow either configuration. Maybe it's worth documenting these possibilities in the README to avoid user confusion?

File: charts/trino/templates/ingress.yaml
Comment (bitsondatadev) [inline charts/trino/templates/ingress.yaml:4]: @hashhar should we consider only supporting greater than not supported by Kubernetes, as per [the Trino documentation](https://trino.io/docs/current/installation/kubernetes.html#requirements).

Comment (bitsondatadev) [reply charts/trino/templates/ingress.yaml:4]: Therefore, we should be using the latest api.

```
apiVersion: networking.k8s.io/v1
```

File: charts/trino/templates/ingress.yaml
Comment (bitsondatadev) [reply charts/trino/templates/ingress.yaml:12]: @yanyixing could you please update this version.

File: charts/trino/templates/ingress.yaml
Comment (bitsondatadev) [inline charts/trino/templates/ingress.yaml:27]: This should be the default now that we're asserting kubernetes >=1.23

File: charts/trino/values.yaml
Comment (bitsondatadev) [inline charts/trino/values.yaml:154]: All supported version use
```
ingressClassName: {{ .Values.ingress.className }}
```
rather than an annotation so let's remove this.

File: charts/trino/values.yaml
Comment (bitsondatadev) [inline charts/trino/values.yaml:160]: Just comment this out to avoid confusion around default values.

```suggestion
#  hosts:
#    - host: trino.example.com
#      paths:
#        - path: /
#          pathType: ImplementationSpecific
```

**Review by bitsondatadev — CHANGES_REQUESTED** [review]
We should avoid supporting any version outside of the k8s support window. See more info here:

https://www.youtube.com/watch?v=SObTY-Ce1G4

**Comment by ckdarby** [comment]
The other comment, in Charts.yaml we'll want to use:

```YAML
kubeVersion: A SemVer range of compatible Kubernetes versions (optional)
```

File: charts/trino/templates/ingress.yaml
Comment (yanyixing) [reply charts/trino/templates/ingress.yaml:16]: updated

File: charts/trino/values.yaml
Comment (yanyixing) [reply charts/trino/values.yaml]: updated

**Comment by yanyixing** [comment]
@bitsondatadev Thanks for your review and suggestions, I have updated the pr, please review again.

File: charts/trino/templates/ingress.yaml
Comment (bitsondatadev) [inline charts/trino/templates/ingress.yaml:39]: @yanyixing lets get rid of this as well.

File: charts/trino/templates/ingress.yaml
Comment (bitsondatadev) [inline charts/trino/templates/ingress.yaml:15]: Remove this if as it is required

File: charts/trino/templates/ingress.yaml
Comment (bitsondatadev) [inline charts/trino/templates/ingress.yaml:35]: Let's remove this

**Review by bitsondatadev — CHANGES_REQUESTED** [review]
Let's remove all version comparisons assuming that people are using the supported k8s versions.

File: charts/trino/templates/ingress.yaml
Comment (yanyixing) [reply charts/trino/templates/ingress.yaml]: removed

**Review by bitsondatadev — APPROVED** [review]
Looks good barring one open question I have to be addressed by Ashhar

**Comment by bitsondatadev** [comment]
Update: Ashhar is just getting back from leave so he'll need a week or so to catch up on things. Just wanted to keep y'all updated since this one has been touch and go.

**Comment by bitsondatadev** [comment]
@hashhar can you take a look here please?

**Comment by bitsondatadev** [comment]
Also tagging @losipiuk since you have some k8s experience :)

**Comment by yanyixing** [comment]
@bitsondatadev It seems like this repo is not updated for a long time

**Comment by bitsondatadev** [comment]
Im actively looking for a maintainer. @hashhar doesn't currently have the bandwidth.

**Comment by mosabua** [comment]
Could you resolve conflicts?

**Pull request merged**
